src/main/java/org/kaazing/netx/ws/internal/io/WsInputStream.java

```diff
@@ -141,101 +145,106 @@ public int available() throws IOException {
 
     @Override
     public int read() throws IOException {
```

Comment:
Are we reading 1 byte at a time and relying on the super class to fill the buffer?
That would imply we're acquiring and releasing the lock for every byte, right?